### PR TITLE
Implements flush and flush_all

### PR DESCRIPTION
**Pull request by nova77**
Apparently the fastest way to clear a leveldb instance is to close the db, delete the directory and start over.

### CODE REVIEW

File: src/queue/queue.cpp [hunk 1]
```diff
@@ -197,6 +211,23 @@ void queue::erase_chunks(const header_type& header)
    write(batch);
 }
 
+void queue::flush_db()
+{
+   log::INFO("queue <%1%>: flushing db", path_);
+
+   wake_up(); // in case there's a waiter waiting
+
+   waiters_.clear();
```
Comment (erikfrey) [inline src/queue/queue.cpp]: hmm, why are you clearing out the waiters?  you're invalidating iterators that are sitting out in asio wait events.

**Comment by erikfrey** [comment]
Hmm, unfortunately this isn't going to work as is.  Queue has a few methods that the client chains together to maintain state.  So, for example, a push with a large object may in fact be multiple pushes.  Something like this.

``` c++
while (chunk = read_chunk())
   queue_.push(chunk);
```

Similarly, popping has a few states:

``` c++
id = queue_.pop_begin();
while (chunk = queue.pop_read(id))
   write_chunk(chunk);
queue.pop_end(id);
```

We need to describe behavior for what happens if a client is in the middle of pushing or popping when you call `flush()`

OR we need to make it illegal to flush during pushing/popping, which would be okay, I suppose

OR we need to make it that flushing can only happen completely in between pushing and popping, but this could be trickier.

File: src/queue/queue.cpp (hunk 1 above)
Comment (nova77) [inline src/queue/queue.cpp]: Well, the idea was to get around the issue you lay below, but it seems I did not know enough about the code.

**Comment by nova77** [comment]
> Queue has a few methods that the client chains together to maintain state

Can you make me an example of such commands chain? I had the impression that the set/get operations were considered atomic from the client prospective.

If we can figure out a way give a state to the chain, we could enforce exclusivity of the flush operation via a shared mutex.

**Comment by nova77** [comment]
Closed in favour of #22.

**Pull request closed without merging**